lib/database_cleaner/active_record/base.rb

```diff
@@ -16,6 +16,9 @@ def self.config_file_location
     class Base < DatabaseCleaner::Strategy
       def self.migration_table_name
         ::ActiveRecord::SchemaMigration.table_name
+      rescue NoMethodError # Rails 7.1 the class was refactored
+        # https://github.com/rails/rails/pull/45908#issuecomment-1403294948
+        ::ActiveRecord::SchemaMigration.new(1).table_name
```

Comment:
That's not how you would typically establish compatibility with different rails versions. You would rather check the version explicitly and split the code path based on that:

```ruby
        if Gem::Version.new("6.0.0") <= ::ActiveRecord.version
          ::ActiveRecord::Base.connection.schema_migration.table_name
        else
          ::ActiveRecord::SchemaMigration.table_name
        end
```